apps/web/src/app/(protectedPage)/forms/[formId]/_components/formEditor/formEditorCard.tsx

```diff
@@ -116,11 +129,20 @@ export function FormEditorCard({ form }: Readonly<Props>) {
   });
   const isFormBusy = updateForm.isPending;
 
-  const onSubmit = (formData: FormSubmitDataSchema) =>
+  const onSubmit = (formData: FormSubmitDataSchema) => {
+    console.log("form data", formData);
+    const cleanedData = {
+      ...formData,
+      formFields: formData.formFields.filter(
+        (field) => field.isMarkedForRemoval != true,
```

Comment:
We can filter fields which are marked to remove by using `(field, index) => !removeFieldsIds.includes(fields[index]!.id),`